### PR TITLE
Adding object files for articles and teams

### PR DESCRIPTION
**Pull request by je-collins**

### CODE REVIEW

**Comment by QuasarCatfish** [comment]
## add_member.js ##
- 2: you import "Teams.js" instead of "Team.js"
- 41: Use "!==" instead of "!=" for safety
- add: A check to make sure user is administrator

## delete_member.js ##
- 2: you import "Teams.js" instead of "Team.js"
- add: A check to make sure user is administrator

Since these two methods are so similar, you can probably merge them into "manage_member.js" with an "add" parameter

## get_articles.js ##
- 2: you import "Article" instead of "Article.js"
- 18: team_id will always be undefined, so this endpoint will always return 400
- 41: articles should never be null, so a null check is redundant

## Article.js ##
- 5: fromId is a bad name for this method, since it returns the first article from the given user; if this is meant to return all articles from a user, choose "getArticlesFromUser" and see 12 for return statement
- 10: should be called getFavoritesFromUser, since it returns a list given the user; id is ambiguous
- 11: "&" in SQL should be "AND"
- 12: should be "return res.length === 0 ? [] : res;"
- 15: should be "getArticles"
- 16: comma should not be present in SQL
- 17: see 12
- 20: should be "getStarredArticlesFromUser"
- 22: see 12
- 25: should be "getStarredArticles"
- 27: see 12
- 30: should be "getStarredDomainsFromUser"
- 32: see 12
- 35: should be "getStarredDomains"
- 37: see 12
- 58: see 12
- 63: see 12

## Team.js ##
- 6: should be "getTeamUserFromId"
- 27: query will fail if the team has any members in it
- 34: see Article.js 12

## server.js ##
- did not add "get_articles" endpoint

File: train_trax/server/endpoints/add_member.js [hunk 1]
```diff
@@ -34,11 +34,18 @@ export default async (request, response) => {
 		return response.status(400).json(json);
 	}
 
+	// If user is not an admin, return bad request
+	if (user.administrator) {
```
Comment (QuasarCatfish) [inline train_trax/server/endpoints/add_member.js]: This checks if the user is an administrator, not if they are not.

File: train_trax/server/endpoints/add_member.js [hunk 2]
```diff
@@ -34,11 +34,18 @@ export default async (request, response) => {
...
+        json.error = 'Bad request';
+		json.message = 'The user is not an admin and connot edit teams.';
+		return response.status(400).json(json);
```
Comment (QuasarCatfish) [inline train_trax/server/endpoints/add_member.js]: Request forbidden is status 403, not 400.

File: train_trax/server/endpoints/delete_member.js [hunk 3]
```diff
@@ -34,6 +34,13 @@ export default async (request, response) => {
 		return response.status(400).json(json);
 	}
 
+	// If user is not an admin, return bad request
+	if (user.administrator) {
```
Comment (QuasarCatfish) [inline train_trax/server/endpoints/delete_member.js]: This checks if the user is an admin, not if they aren't.

File: train_trax/server/endpoints/delete_member.js [hunk 4]
```diff
@@ -34,6 +34,13 @@ export default async (request, response) => {
...
+        json.error = 'Bad request';
+		json.message = 'The user is not an admin and connot edit teams.';
+		return response.status(400).json(json);
```
Comment (QuasarCatfish) [inline train_trax/server/endpoints/delete_member.js]: 403 not 400

File: train_trax/server/endpoints/delete_member.js [hunk 5]
```diff
@@ -34,6 +34,13 @@ export default async (request, response) => {
...
+	}
+
     // Check if the user exists in the team
     const member = await Team.fromId(team_id, member_id);
```
Comment (QuasarCatfish) [inline train_trax/server/endpoints/delete_member.js]: Function name was changed to getTeamUserFromId

**Review by QuasarCatfish — APPROVED** [review]

**Comment by je-collins** [comment]
Added the following endpoints
- add to team
- delete from team
- search library

**Pull request merged**